lib/rubygems/spec_fetcher.rb

```diff
@@ -171,30 +171,34 @@ def spec_for_dependency(dependency, matching_platform=true)
   # Suggests gems based on the supplied +gem_name+. Returns an array of
   # alternative gem names.
 
-  def suggest_gems_from_name(gem_name, type = :latest)
+  def suggest_gems_from_name(gem_name, type = :latest, num_results = 5)
     gem_name        = gem_name.downcase.tr('_-', '')
+    name            = Regexp.new gem_name
     max             = gem_name.size / 2
     names           = available_specs(type).first.values.flatten(1)
 
     matches = names.map do |n|
       next unless n.match_platform?
-
-      distance = levenshtein_distance gem_name, n.name.downcase.tr('_-', '')
-
-      next if distance >= max
-
-      return [n.name] if distance == 0
-
-      [n.name, distance]
+      [n.name, 0] if name === n.name.downcase.tr('_-', '')
```

Comment:
Any reason to prefer regular expressions and `===` over this? To the best of my knowledge, they're equivalent.

```suggestion
      [n.name, 0] if n.name.downcase.tr('_-', '').include?(gem_name)
```